### ISSUE TITLE
optimum inference for summarization

### ISSUE THREAD

**Issue by ShilpaSangappa** [huggingface/optimum#178]
With reference to the blog: https://huggingface.co/blog/optimum-inference, I am able to do this:

```
from transformers import AutoTokenizer, pipeline
-from transformers import AutoModelForQuestionAnswering
+from optimum.onnxruntime import ORTModelForQuestionAnswering

-model = AutoModelForQuestionAnswering.from_pretrained("deepset/roberta-base-squad2") # pytorch checkpoint
+model = ORTModelForQuestionAnswering.from_pretrained("optimum/roberta-base-squad2") # onnx checkpoint
tokenizer = AutoTokenizer.from_pretrained("deepset/roberta-base-squad2")

optimum_qa = pipeline("question-answering", model=model, tokenizer=tokenizer)

question = "What's my name?"
context = "My name is Philipp and I live in Nuremberg."
pred = optimum_qa(question, context)
```
I need to do similar inference for summarization, for the following code:

```
from transformers import AutoTokenizer, AutoModelForSeq2SeqLM
tokenizer = AutoTokenizer.from_pretrained("sshleifer/distilbart-cnn-12-6")
model = AutoModelForSeq2SeqLM.from_pretrained("sshleifer/distilbart-cnn-12-6")

```
I get the following error:
```
>>> from optimum.onnxruntime import ORTModelForSeq2SeqLM
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'ORTModelForSeq2SeqLM' from 'optimum.onnxruntime' (/datadrive/shilpa/work/virtual_environments/venv_hf_optimum/lib/python3.9/site-packages/optimum/onnxruntime/__init__.py)
```
https://huggingface.co/docs/optimum/main/en/pipelines - mentions "text-generation" as one of the supported tasks,  "summarization"  i assumed comes under this category. Am I right?

**Comment by fxmarty** [comment]
Thank you for using Optimum! Currently, the supported model class for text-classification looks to be `ORTModelForCausalLM`, see:

https://github.com/huggingface/optimum/blob/6aa47f96be4644acdfbaa2a006b891ae1bad5474/optimum/pipelines.py#L58-L62

Interestingly, the transformers' `AutoModelForSeq2SeqLM` is loading at the end of the day `...ForConditionalGeneration`, see [here](https://github.com/huggingface/transformers/blob/0e6ec2a4698916ba8f26272564c1776fdabc89cf/src/transformers/models/auto/modeling_auto.py#L753) and [here](https://github.com/huggingface/transformers/blob/63517fdf4840505425c5f516ade17112dc769ebf/src/transformers/models/auto/modeling_auto.py#L395). Looks like this is not yet supported in Optimum.

However, looks like the `text-generation` task is mapped to `...ForCausalLM` models, see [here](https://github.com/huggingface/transformers/blob/main/src/transformers/modelcard.py#L63) and [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/auto/modeling_auto.py#L232).

**Comment by philschmid** [comment]
Hey @ShilpaSangappa, 

We currently only have support `text-generation` or rather speaking Decoder models like `GPT-2`, `GPT-Neo`, `GPT-J`. But we are working on adding support for Seq2Seq models to be able to use Encoder-Dedocer models and tasks like `summarization` and `translation`.

**Comment by ShilpaSangappa** [comment]
@fxmarty @philschmid Thank you for the clarifications.

**Comment by ShilpaSangappa** [comment]
@fxmarty @philschmid Thank you for the clarifications.

**Comment by philschmid** [comment]
@Ierezell sorry not yet. I ll try to update this issue if we have concrete timeline for it.

**Comment by echarlaix** [comment]
@ShilpaSangappa @Ierezell the `text2text-generation`, `summarization` and `translation` pipelines are finally supported as we merged #199
